streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -464,9 +463,7 @@ void handleLostAll() {
             // Even though we've apparently dropped out of the group, we can continue safely to maintain our
             // standby tasks while we rejoin.
             if (task.isActive()) {
-                cleanupTask(task);
-                task.prepareCloseDirty();
-                task.closeDirty();
+                closeTaskDirty(task);
```

Comment:
Should we also close producers for the lost tasks as well?

Reply:
We do, right below this, but rather than delegate to `#cleanupTaskProducer`  we just call `activeTaskCreator.closeAndRemoveTaskProducerIfNeeded` directly (which is what `#cleanupTaskProducer` ultimately does). I think we should consolidate these calls eventually, but not necessarily in this hotfix